### PR TITLE
TES-381 Added provisioning of graphdb backup script

### PR DESCRIPTION
**Pull request by viktor-ribchev**
## Description
Added installation of Telegraf.
Added provisioning of GraphDB backup script.
Added keepalive and file max size settings.
Improved logging of the GraphDB install script.

## Related Issues

TES-381

## Changes

Added installation of Telegraf.
Added provisioning of GraphDB backup script.
Added keepalive and file max size settings.
Improved logging of the GraphDB install script.

## Screenshots (if applicable)

N/A

## Checklist

- [x] I have tested these changes thoroughly.
- [x] My code follows the project's coding style.
- [x] I have added appropriate comments to my code, especially in complex areas.
- [x] All new and existing tests passed locally.

### CODE REVIEW

File: files/graphdb_backup [hunk 1]
```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+# This script utilizes the cloud backup functionality of GraphDB.
+# It is intended to be run in clustered topology.
+
+set -euo pipefail
+
+az login --identity
+
+RESOURCE_GROUP="$(curl -s -H Metadata:true "http://169.254.169.254/metadata/instance/compute/resourceGroupName?api-version=2021-01-01&format=text")"
```
Comment (mihailradkov) [inline files/graphdb_backup]: I don't like these variables here. I could use the image with another user data script when creating VMs, we should not restrict this, IMO. Also, the way how the script assumes that the first storage account and app config are the ones to be used... it's wrong as it is not portable to most subscriptions. Better to pass the variables as arguments.

File: variables.pkr.hcl [hunk 2]
```diff
@@ -1,85 +1,116 @@
...
-  type = string
+  description = "Client secret of the service principal"
+  type        = string
+  default     = null
+}
+
+variable use_azure_cli_auth {
+  description = "CLI auth will use the information from an active az login session to connect to Azure and set the subscription id and tenant id associated to the signed in account."
+  type        = bool
+  default     = false
```
Comment (mihailradkov) [inline variables.pkr.hcl]: Let's make this `true` by default?

File: files/graphdb_backup [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+# This script utilizes the cloud backup functionality of GraphDB.
+# It is intended to be run in clustered topology.
+# This script accepts three variables in the following order: app_config_name, backup_storage_account_name, backup_storage_container_name
+
+set -euo pipefail
+
+az login --identity
+
+GRAPHDB_ADMIN_PASSWORD="$(az appconfig kv show --name ${1} --auth-mode login --key graphdb-password | jq -r .value | base64 -d)"
```
Comment (mihailradkov) [inline files/graphdb_backup]: Can't we provide it? It's coupling the script to KV.

**Review by mihailradkov — COMMENTED** [review]
Update the PR description to mention the extra commits

**Review by yaskoo — APPROVED** [review]

File: files/graphdb_backup [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+      ((retry_count=retry_count + 1))
+      sleep 5
+    fi
+  done
+}
+
+# Checks if GraphDB is running in cluster
+    IS_CLUSTER=$(
+      curl -s -o /dev/null \
```
Comment (mihailradkov) [inline files/graphdb_backup]: Why the indent?

**Review by mihailradkov — APPROVED** [review]

**Review by yaskoo — APPROVED** [review]

File: files/graphdb_backup (hunk 1 above)
Comment (viktor-ribchev) [reply files/graphdb_backup]: done

File: variables.pkr.hcl (hunk 2 above)
Comment (viktor-ribchev) [reply variables.pkr.hcl]: done

File: files/graphdb_backup (hunk 3 above)
Comment (viktor-ribchev) [reply files/graphdb_backup]: done

File: files/graphdb_backup (hunk 4 above)
Comment (viktor-ribchev) [reply files/graphdb_backup]: fixed

**Pull request merged**